nxdrive/engine/processor.py

```diff
@@ -408,10 +408,10 @@ def _execute(self) -> None:
     def _handle_pair_handler_exception(
         self, doc_pair: DocPair, handler_name: str, e: Exception
     ) -> None:
-        if isinstance(e, OSError) and e.errno == errno.ENOSPC:
+        if isinstance(e, OSError) and e.errno in NO_SPACE_ERRORS:
             self.engine.suspend()
-            log.warning("No space left on device!")
-            self.increase_error(doc_pair, "NO_SPACE_LEFT_ON_DEVICE")
+            log.warning("No space left on device!", exc_info=True)
+            self.engine.get_queue_manager().push(doc_pair)
```

Comment:
Maybe it is better to keep the old behavior (pushing the doc in error)?

Reply:
I think the new behavior is good. It's not the document's fault that there is no space left, so it doesn't need to be in error.